client/components/auto-direction/index.jsx

```diff
@@ -37,9 +77,18 @@ const getContent = ( reactElement ) => {
 	// This child has it's content set to external HTML
 	if ( typeof props.dangerouslySetInnerHTML === 'object' ) {
 		// Strip tags because we're only interested in the text, not markup
-		return props.dangerouslySetInnerHTML.__html
-			? stripHTML( props.dangerouslySetInnerHTML.__html.substring( 0, Math.floor( MAX_LENGTH_OF_TEXT_TO_EXAMINE * 1.25 ) ) )
-			: '';
+		// We examine from the first position without tags of the string, so we won't get an empty text
+		// because we might get only tags in the beginning
+		const html = props.dangerouslySetInnerHTML.__html;
+		if ( ! html ) {
+			return '';
+		}
+
+		const examinedLength = Math.floor( MAX_LENGTH_OF_TEXT_TO_EXAMINE * 1.25 );
+		const taglessIndex = getTaglessIndex( html );
+		const startIndex = taglessIndex + examinedLength < html.length ? taglessIndex : 0;
```

Comment:
I'm confused by the relationship between `examinedLength` and `startIndex`.

1. what does examinedLength represent? Why are we essentially calculating `125` over and over instead of changing the constant?
2. what is the purpose of this line: `const startIndex = taglessIndex + examinedLength < html.length ? taglessIndex : 0`?
3. Do we have any unit tests for `AutoDirection`? If not can we add some? I think this component definitely warrants some :).  To start it can be just the basics (tagful text, rtl, ltr)